### ISSUE TITLE
Quality difference vs official repo & ~2s silent audio with weak intro (GGUF BF16, 24GB)

### ISSUE THREAD

**Issue by c0asters** [ServeurpersoCom/acestep.cpp#105]
Thank you for this project — your work has been really helpful. I do have a couple of pieces of feedback, though. I'm running the GGUF BF16 quantized build and it fits comfortably within 24GB of VRAM.

1、Compared to the official ACE-Step 1.5 repo, using the same 4B LM and XL-TURBO models, the generated songs feel slightly lower in quality than what the official repo produces. I'm not sure if this is related to hyperparameters — the official repo exposes a lot of fine-grained parameter tuning that this build doesn't seem to.

2、Another issue: when the intro / lead-in signal is weak, there's about 2 seconds of silent audio at the beginning. Since BF16 should have nearly identical precision to the original weights, the model behavior should be consistent — so I'm wondering whether this gap is caused by differences between the GGML and the Python implementations?

**Comment by ServeurpersoCom** [comment]
Thanks for the detailed report.

I ran a full stage by stage comparison against the official repo on XL-TURBO, BF16 GGUF against the PyTorch reference in bfloat16, same seed and the same audio codes fed to both sides. The harness is in this repo, tests/debug-dit-cossim.py, it dumps every stage and prints cosine similarities. Text encoder, lyric encoder, detokenizer, noise, timestep embedding and the DiT trunk all agree to four or five nines, and what is left over the sampler trajectory is ordinary bf16 accumulation. The math is not where your difference comes from.

Sampling parameters are, and you are right that the official repo exposes more of them. The one that matters most: the official pipeline turns DCW on by default, mode double, scaler 0.05, high_scaler 0.02, haar wavelet, while here it stays off unless the request asks for it. Add "dcw_mode": "double", "dcw_scaler": 0.05 and "dcw_high_scaler": 0.02 to your request and you get the same correction. Keep the scalers that small, larger values degrade the result without warning.

For the rest I would rather not guess. Both codebases are open, the reference is at https://github.com/ace-step/ACE-Step-1.5, and a frontier model like Claude or GPT (the two I use) will happily read both samplers side by side and tell you which parameter is not wired the same way. If you find one, post it here with the detail, the parameter and where the two paths diverge, and I will look at it.

On the two seconds of silence I need pairs to work from: the prompt, the audio the official repo produced, and the audio this port produced, same seed and same settings. I cannot reproduce it blind. Worth knowing that the intro is planned by the LM, not the DiT, so the codes it emits are what decide whether a track opens on silence.

For what it is worth my own listening goes the other way, I find the GGML path and the C++ post processing more precise than the reference. Which is exactly why I want your files.

**Comment by ServeurpersoCom** [comment]
DCW off on both sides, then on with the official defaults (double, 0.05, 0.02, haar):

```
                 DCW off     DCW on
dit_step0_xt     0.999843    0.999930     mean_err 0.0124 -> 0.0086
dit_step4_xt     0.992773    0.996629     mean_err 0.0720 -> 0.0508
dit_step6_xt     0.962166    0.968277     mean_err 0.1667 -> 0.1462
```

Cosine similarity: 1.0 means identical, so this is bf16 rounding and not a bug, and a 0.2 gap on that scale is nothing a human ear picks out.

**Comment by pillopaus-project** [comment]
i honestly think that acestep .cpp results are better than the online  Ace. i tried a few generations with the same json and  imho they are not in the same league...
this is a released example of acestep use in my workflow : 
https://pillopaus-project.bandcamp.com/track/longer-night-deep-house-full-vocal-mix
i still have not heard this kind of results from the online  original Ace.
as always all IMHO.
PP

**Comment by ServeurpersoCom** [comment]
> i honestly think that acestep .cpp results are better than the online Ace. i tried a few generations with the same json and imho they are not in the same league... this is a released example of acestep use in my workflow : https://pillopaus-project.bandcamp.com/track/longer-night-deep-house-full-vocal-mix i still have not heard this kind of results from the online original Ace. as always all IMHO. PP

I listened to your tracks I like electronic/house also, Acestep's turbo models are very fast but have a slight spectral bias that you can correct using DCW, you need very low values, and while it doesn't eliminate it completely, it helps. You might also want to try https://github.com/ServeurpersoCom/yue2.cpp, which seems better than minimaxmusic.cpp for electronic music; it offers "a good latent space to explore" :)

YuE2 is the very first to offer score/melody control and repeatability across generations based on a catchy theme.

**Comment by pillopaus-project** [comment]
what i like of acestep is how it preserves the character of  my analog synths when they are given as ref-audio. it let me swap the entire palette of sounds faster then re-finding the sweet spot for every sound on the no patch memory/ no midi params synths i have. now that i have a varied palette of vae ref profiles it's a great workflow for its use cases.. 
im trying yue now .. looks promising for my kind of workflow as i could give him the midi  to follow instead of going through the cover-nofsq  at low val. im using now that gives good results but can be time consuming.
thanks for the great work there too ! 
PP
